content/articles/federated-learning-using-tensorflow-federated/index.md

```diff
@@ -0,0 +1,201 @@
+
+1. TFF is architecture-agnostic. 
+
+This means that it is able to compile all code into an abstract representation. Its abstract representation nature enables it to be deployed in diverse environments.
+
+2. TFF saves effort
+
+It is designed to mitigate the pain points that we developers face when developing our own federated learning systems. Some of these pain points include interleaving the different types of logic, the global vs local perspective on communication, and tension between the order of construction vs. execution.
+
+3. TFF has many extensions.
+
+Differential privacy, compression, and quantization are some of the few extensions available in TFF.
+
+#### TensorFlow federated layers
+
+TFF offers two main layers:
+
+1. Federated Learning (FL) API 
+
+The FL API is a high-level API that implements federated training and evaluation. It can be applied to your existing TensorFlow models or data.
+
+2. Federated Core (FC) API
+
+The FC API is much more lower-level and sits below the FC API. This API gives you more generic expressions that allow you to run and simulate custom types of computations and control your own orchestrations. It also has a local runtime which allows for simulations.
+
+> In this tutorial, we will only be focusing on the FL API and the code behind it. This is to avoid the tutorial becoming too long. We will tackle the FC API in my next article.
+
+#### Ways to get involved
+
+There are different ways you can get involved depending on your interest:
+
+1. ML Devs
+
+ML devs help in applying Federated Learning APIs to existing TensorFlow models.
+
+2. FL researchers
+
+This type of researchers helps to design new federated learning algorithms using the FC API.
+
+3. Systems researchers
+
+These types of researchers help in optimizing generated computation structures.
+
+4. System Devs
+
+These developers help integrate TFF with new types of development environments.
+
+### The code behind TensorFlow Federated (TFF)
+
+First, let's briefly take a look at how the Keras model looks like:
+
+```python
+    def create_compiled_keras_model():
+        model = tf.keras.models.Sequential([
+            tf.keras.layers.Dense(
+                10, activation=tf.nn.softmax, kernel_initializer = 'zeros', input_shape = (784, 
+                )
+            )
+        ])
+
+        model.compile(
+            loss=tf.keras.losses.SparseCategoricalCrossentropy(),
+            optimizer=tf.keras.optimizers.SGD(learning_rate=0.02),
+            metrics=[tf.keras.metrics.SparseCategoricalAccuracy()] 
+
+        )
+    return model
+```
+
+If you're familiar with TensorFlow, the above Keras code already looks familiar to you. We will import it into our main function `model_fn` using the `create_compiled_keras_model()` method.
```

Comment:
You assume a reader is familiar with the code in this section however a reader who has met all the prerequisites to this tutorial may derive more value if you give some form of implementation context of this Keras model. i.e. Why a sequential model? When/for what purpose do we typically use such a model? What is its significance here?
This might also lock out a curious reader who may want to understand this step by step